src/chapters/git/branches.rst

```diff
@@ -0,0 +1,248 @@
+Branching Out
+=============
+
+What if we want to experiment with a new feature, and we know that we will need
+to make several commits before its done? Similarly, what if two students want
+to work on different parts of a project at the same time?
+
+For example, one student could work on some HTML code while their partner
+completes the CSS. They can both work during class to save themselves some
+time. However, their commits won't follow the same path. Each person will have
+their own version of the Git history for the same project. As we saw on the
+previous page, this is not a good idea.
+
+Unless...
+
+.. figure:: figures/git-branch-intro.png
+   :alt: The HTML and CSS code can be split from the main program.
+   :width: 70%
+
+   Branches to the rescue.
+
+Not only does Git allow us to store the version history of our project, it also
+lets us split the development into different paths.
+
+.. index:: ! branch
+
+.. index::
+   single: git; branch
+
+Branching in Git
+----------------
+
+A Git **branch** is a separate version of the same ``main`` code. Like a branch
+on a tree, a branch in Git shares the same trunk as the other limbs. However,
+each branch is distinct from all of the others.
+
+With branches, the two students can work on *separate* versions of the same
+project. They won't need to worry about messing up each other's work every time
+they want to make a new commit. They share the same base code, but they update
+their own parts of it. Eventually, they will merge their separate branches back
+into the main program.
+
+Branches let programmers start, update, store, and test new features for their
+code without disrupting the main flow of the Git history.
+
+On Branch ``main``
+------------------
+
+In the :ref:`previous section <check-git-status>`, we used ``git status`` to
+check the state of the ``git_practice`` repository. The first line in the
+output was ``On branch main``.
+
+.. admonition:: Note
+
+   If your installed version of ``git`` gives the default branch a name other
+   than ``main``, you can rename it with the command:
+
+   ::
+
+      $ git branch -m old-branch-name main
+
+   In this case, ``old-branch-name`` becomes ``main``.
+
+When we run ``git init`` to start a new repo, part of the process creates a
```

Comment:
Yes, Git still uses `master`, and GitHub uses `main`.  To avoid conflicts later in the chapter, I wanted to deal with renaming the default branch early.